### PR TITLE
K8s CRD definition for Sentinel rules

### PR DESCRIPTION
**Pull request by louyuting**
### Special notes for reviews

### CODE REVIEW

File: ext/datasource/k8s/deploy/crds/datasource.sentinel.com_circuitbreakerrules_crd.yaml [hunk 1]
```diff
@@ -0,0 +1,92 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: cbrules.datasource.sentinel.com
```
Comment (sczyh30) [inline ext/datasource/k8s/deploy/crds/datasource.sentinel.com_circuitbreakerrules_crd.yaml:4]: The abbr might not be clear enough. Maybe cbrules -> `circuitbreaker.rules` or `circuitbreakerrules` or other better representation?

Comment (sczyh30) [reply ext/datasource/k8s/deploy/crds/datasource.sentinel.com_circuitbreakerrules_crd.yaml:4]: In open-source world we'd like `*.sentinel.io` (though the domain is not owned by us :(

File: ext/datasource/k8s/deploy/crds/datasource.sentinel.com_flowrules_crd.yaml [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+                    format: int32
+                    type: integer
+                    minimum: 0
+                  maxQueueingTimeMs:
+                    format: int32
+                    type: integer
+                    minimum: 0
+                required:
+                - resource
+                - limitOrigin
```
Comment (sczyh30) [inline ext/datasource/k8s/deploy/crds/datasource.sentinel.com_flowrules_crd.yaml:79]: Only resource, metricType and count are required. For other attributes, we could provide default values.

File: ext/datasource/k8s/deploy/crds/datasource.sentinel.com_hotspotrules_crd.yaml [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+                      - specificThreshold
+                      type: object
+                    type: array
+                required:
+                - resource
+                - metricType
+                - controlBehavior
+                - paramIndex
+                - threshold
+                - maxQueueingTimeMs
```
Comment (sczyh30) [inline ext/datasource/k8s/deploy/crds/datasource.sentinel.com_hotspotrules_crd.yaml:107]: Please also simplify the required list here.

File: ext/datasource/k8s/deploy/crds/datasource.sentinel.com_circuitbreakerrules_crd.yaml (hunk 1 above)
Comment (louyuting) [reply ext/datasource/k8s/deploy/crds/datasource.sentinel.com_circuitbreakerrules_crd.yaml:4]: I prefer circuitbreaker.rules

File: ext/datasource/k8s/deploy/crds/datasource.sentinel.com_flowrules_crd.yaml (hunk 2 above)
Comment (louyuting) [reply ext/datasource/k8s/deploy/crds/datasource.sentinel.com_flowrules_crd.yaml:79]: make sense

File: ext/datasource/k8s/deploy/crds/datasource.sentinel.com_hotspotrules_crd.yaml (hunk 3 above)
Comment (louyuting) [reply ext/datasource/k8s/deploy/crds/datasource.sentinel.com_hotspotrules_crd.yaml:107]: Okay

**Comment by louyuting** [comment]
Duplicated #242

**Pull request closed without merging**